File: company-application/src/main/java/telran/employees/WageEmployee.java
package telran.employees;

import java.time.LocalDate;

public class WageEmployee extends Employee {
    private int wage;
    private int hours;

    public WageEmployee(long id, String name, String department, int wage, int hours) {
        super(id, name, department, wage * hours, LocalDate.now());
        this.wage = wage;
        this.hours = hours;
    }

    @Override
    public String toString() {
        return super.toString() + ", WageEmployee{" +
                "wage=" + wage +
                ", hours=" + hours +
                '}';
    }
}

File: Node-Project/company/src/dto/WageEmployee.mjs
import Employee from "./Employee.mjs";
export default class WageEmployee extends Employee{
    static {
        Employee.classMap.WageEmployee = new WageEmployee();
    }
    constructor(id, department, basicSalary,  wage = 0, hours=0, className) {
        super(id, department, basicSalary, className ?? "WageEmployee");
        this.wage = wage;
        this.hours = hours;
    }
    getWage() {
        return this.wage;
    }
    getHours() {
        return this.hours
    }
    computeSalary () {
        return super.computeSalary() + this.hours * this.wage
    }
}